### PR TITLE
Trunk-5480 Make ModuleUtil support more version specification format

### PR DESCRIPTION
**Pull request by Andu033**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5480

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by Andu033** [comment]
@mks-d   Can you review it please? I know that the documentation changes are not very good but i dont know how to get it right. Thanks a lot.

File: api/src/test/java/org/openmrs/module/ModuleUtilTest.java [hunk 1]
```diff
@@ -187,6 +187,72 @@ public void matchRequiredVersions_shouldAllowRequiredVersionWithWildCard() {
...
 	}
 	
+	/**
+	 * @see org.openmrs.module.ModuleUtil#matchRequiredVersions(String,String)
+	 */
+	@Test
+	public void matchRequiredVersions_shouldAllowRequiredVersionWithPlus() {
+		String openmrsVersion = "1.4.3";
+		String requiredOpenmrsVersion = "1.4.2+";
+		Assert.assertTrue(ModuleUtil.matchRequiredVersions(openmrsVersion, requiredOpenmrsVersion));
```
Comment (mks-d) [inline api/src/test/java/org/openmrs/module/ModuleUtilTest.java]: Please test the edges. So in this case values such as: "1.4.1", "1.4.1.99", "1.4.2", "1.4.3", "1.5.0", ... etc.
You don't need to reassign `openmrsVersion` each time then, I'm good with something like:
```java
String requiredOpenmrsVersion = "1.4.2+";
Assert.assertFalse(ModuleUtil.matchRequiredVersions("1.4.1", requiredOpenmrsVersion));
Assert.assertTrue(ModuleUtil.matchRequiredVersions("1.4.2", requiredOpenmrsVersion));
Assert.assertTrue(ModuleUtil.matchRequiredVersions("1.4.3", requiredOpenmrsVersion));
...
```

**Comment by Andu033** [comment]
@mks-d  added the tests for edge cases . I hope that i didnt miss something . I wait your review . Thanks  
:smiley_cat:

**Comment by Andu033** [comment]
@mks-d  i think i broke the commit history by trying to squash this branch. How can i fix?

**Comment by mks-d** [comment]
Locally, start off a fresh master and cherry-pick the one commit on top:
```bash
git checkout -b tmp upstream/master
git cherry-pick 0d83e403ccd1a1a332404b97c9701abc3b724c75
# here make sure that you have indeed the latest master PLUS the one commit
git log -20 --oneline --format='%h %<(150,trunc)%s'
# now let's archive the current TRUNK-5480
git branch -m TRUNK-5480 archive
# now let's make tmp the new TRUNK-5480
git branch -m TRUNK-5480
# finally let's push off the new TRUNK-5480
git push -f origin TRUNK-5480
```
I'm assuming that your remote is named `origin`. That's the rough idea, when all is done and fine, get rid of the archive: `git branch -D archive`.

**Comment by Andu033** [comment]
@mks-d it worked like a charm .Thank you very much .

**Comment by Andu033** [comment]
@mks-d  thanks for review . I did the requested changes.

**Comment by Andu033** [comment]
@mks-d i think the PR is almost ready for merge.

**Comment by Andu033** [comment]
@mks-d this pr needs review.
